### DISCUSSION TITLE
Why 2 separate MySQL query for relationship?

### DISCUSSION THREAD

**Discussion by Uttam-EWW** [adonisjs/core#3462] [Help]
Why 2 separate MySQL query for relationship?
`User` Model:
![image](https://user-images.githubusercontent.com/86663119/147899661-60d9a39b-d32c-4f68-b992-b30d473db9ca.png)

`UserGroup` Model:
![image](https://user-images.githubusercontent.com/86663119/147899719-26313a68-beb9-4f85-a22f-a544defd98a3.png)

when I execute `await user.related('inGroups').query().preload('group');` why 2 separate SQL queries are being executed?

![image](https://user-images.githubusercontent.com/86663119/147899680-9bb8a0ad-51e9-4557-bb0a-60c3a420e5fa.png)

**Comment by Uttam-EWW** [comment]
@thetutlage Really sorry for tagging you here but I really need to know why it is happening. Please let us know what's going on here.

**Reply by Barbapapazes** [reply]
Can you explain more what is UserGroup ? Is-this a model to link a user and a group ?

**Reply by Uttam-EWW** [reply]
yes.

**Reply by Barbapapazes** [reply]
First, I think you should take a look to a [many to many relationship](https://docs.adonisjs.com/reference/orm/relations/many-to-many). In fact, UserGroup is just a [pivot table](https://fmhelp.filemaker.com/help/18/fmp/en/index.html#page/FMP_Help/many-to-many-relationships.html) and a migration is enough, no model is needed!

Then, I didn't try but I think related is one and preload is another and [then Lucid will set then on the correct instance!](https://docs.adonisjs.com/reference/orm/query-builder#methodsproperties)

Maybe you can expected only one using many to many relationship!
